documentation/tutorial/extra/async/Async in flow.md

```diff
@@ -0,0 +1,490 @@
+
+### `wait` for a `Future`
+
+`delay` will return a `Void` object, thus no variable declaration is needed. In general, an `ACTOR` would return an object *asynchronously*. In this case, a variable should be used to receive the object. The variable *must* be declared together with the `wait` statement without the`Future` wrapper.
+
+```c++
+ACTOR Future<int> returnInt() {
+    wait(delay(1));
+    return 1;
+}
+
+ACTOR Future<Void> waitInt() {
+    int intValue = wait(returnInt());
+    std::cout << "Received " << intValue << std::endl;
+
+    return Void();
+}
+```
+
+ See `waitInt.actor.cpp` for more details.
+
+### `state` d variable
+
+As the `ACTOR` function will be transpiled into several complicated classes, the variable scope will be affected, and is different from standard C++ function variables. If a variable is defined as `state`, then it is accessible within the *whole* actor, no matter where it is defined. However, its value will be the default value before the first assignment. On the other hand, a variable that is defined without `state` will have limited scope, i.e. between its definition and the `wait` call after it.
+
+```C++
+ACTOR Future<Void> state_1() {
+	// The variable marked as state will be accessible everywhere in the ACTOR
+	state int stateValue = 0;
+
+	// The variable without state decorator will have scope between the
+	// declaration of itself and the first wait call. Thus it is possible to
+	// redefine it after the wait call.
+	int value = 0;
+
+	std::cout << "The stateValue is " << stateValue << " while value is " << value << std::endl;
+
+	wait(delay(1.0));
+
+	int value = -1;
+	++stateValue;
+	std::cout << "The stateValue is " << stateValue << " while value is " << value << std::endl;
+
+	return Void();
+}
+
+ACTOR Future<Void> state_2() {
+	// Even the state variable is defined after
+	std::cout << "State variable defined later can be accessed before its definition: stateVariable = " << stateVariable << std::endl;
+
+	wait(delay(0.1));
+
+	state int stateVariable = 10;
+
+	return Void();
+}
+```
+
+See `state.actor.cpp` for more details.
+
+#### Variable declarations in control blocks
+
+It is important to remember that, if `wait`, or other `ACTOR` commands are used in the control block, then the variables should be `state` decorated. The reason is that, the `ACTOR` compiler would compile the code into different functions inside a class. In this case, the variable scope needs to be class-wide, thus `state` is necessary. Such examples can be seen in the latter part of this article.
+
+#### Add $\lambda$ functions to `ACTOR`
+
+`ACTOR`s will be compiled into C++ classes; thus, a $\lambda$ function inside the `ACTOR` will also affected by the scoping of `ACTOR`s. Additionally, the variable capturing of the $\lambda$ function is usually unexpected. To allow $\lambda$ functions access the `state` variables, it is necessary to explicitly let the $\lambda$ function capture `*this`. For example:
```

Comment:
` To allow $\lambda$ functions access` --> ` To allow $\lambda$ functions to access`